stl/inc/mutex

```diff
@@ -95,9 +95,8 @@ public:
 
     using native_handle_type = void*;

```

Comment:
Should we still declare the `native_handle_type` even if `native_handle` is not usable? I guess this is technically conforming, but looks weird.

Reply:
Yeah, it's an open question.